src/map/hash_trie_map/mod.rs

```diff
@@ -21,6 +21,15 @@ use sequence::list;
 use List;
 use self::sparse_array_usize::SparseArrayUsize;
 
+#[cfg(feature = "serde")]
```

Comment:
We can put everything "serde" in a module (in the same file).  I think it is better in terms of organization and it also avoids these multiple `#[cfg(...)]`

Reply:
So like
```
#[cfg(feature = "serde")]
mod serde_impl {
use super::*;
use serde::ser::{Serialize, Serializer};
use serde::de::{Deserialize, Deserializer, SeqAccess, Visitor};
use std::marker::PhantomData;
use std::fmt;

...

}
```
? Do you want any indentation or re-ordering of `use` statements from that suggestion?

Reply:
Exactly.

> Do you want any indentation or re-ordering of use statements from that suggestion?

Order is fine, but indent everything in the `mod` block